File: src/main/golang/leetcode/editor/cn/[27]Remove Element.go
package lc

//leetcode submit region begin(Prohibit modification and deletion)
func removeElement(nums []int, val int) int {
	l, r := 0, 0
	for r < len(nums) {
		if nums[r] != val {
			nums[l], nums[r] = nums[r], nums[l]
			l++
		}
		r++
	}
	return l
}

//leetcode submit region end(Prohibit modification and deletion)

File: src/main/golang/leetcode/editor/cn/[27]Remove Element.py
from typing import List


# leetcode submit region begin(Prohibit modification and deletion)
class Solution:
    # def removeElement(self, nums: List[int], val: int) -> int:
    #     n = len(nums)
    #     i, j = 0, n - 1
    #     while j >= 0 and nums[j] == val: j -= 1
    #     while i < j:
    #         if nums[i] == val:
    #             nums[i], nums[j] = nums[j], nums[i]
    #             while j >= 0 and nums[j] == val: j -= 1
    #         i += 1
    #     return j + 1
    def removeElement(self, nums: List[int], val: int) -> int:
        n = len(nums)
        i, j = 0, 0
        while j < n:
            if nums[j] != val:
                nums[i], nums[j] = nums[j], nums[i]
                i += 1
            j += 1
        return i
# leetcode submit region end(Prohibit modification and deletion)
